packpack

```diff
@@ -127,7 +142,11 @@ if [ "${OS}" = "alpine" ]; then
         echo "adduser $USER abuild";
         echo "export HOME=/home/$USER" ;
         echo "abuild-keygen -a -i -n";
-        echo "su-exec $USER \$@";
+        if [[ 0${IS_ROOTLESS} == 0 ]]; then
+            echo "su-exec $USER \$@";
+        else
+            echo "exec \$@";
```

Comment:
So the user will be added, but not used? I would not add it in the case. We can either skip `--entrypoint` option at all in the case or generate userwrapper with just `exec $@`. Seems meaningful?

(The same for non-alpine block below.)

Reply:
I am not familiar with `su-exec`; https://devops.stackexchange.com/a/5242 notes that `su-exec` is an Alpine script that sounds like the equivalent of a "`sudo` undo" - run the script as a regular user. Since podman has users running without root anyway, I believe it would still use the user you added. It just calls the standard "replace this process `exec`."

Reply:
I meant the following (I don't test it):

```diff
diff --git a/packpack b/packpack
index b543897..69aada3 100755
--- a/packpack
+++ b/packpack
@@ -133,7 +133,9 @@ esac
 #
 mkdir -p ${BUILDDIR}
 
-if [ "${OS}" = "alpine" ]; then
+if [ "${IS_ROOTLESS}" = "1" ]; then
+    echo "exec \$@" > ${BUILDDIR}/userwrapper.sh
+elif [ "${OS}" = "alpine" ]; then
     (
         echo "#!/bin/sh";
         echo "adduser -u $(id -u) -D $USER";
@@ -142,11 +144,7 @@ if [ "${OS}" = "alpine" ]; then
         echo "adduser $USER abuild";
         echo "export HOME=/home/$USER" ;
         echo "abuild-keygen -a -i -n";
-        if [[ 0${IS_ROOTLESS} == 0 ]]; then
-            echo "su-exec $USER \$@";
-        else
-            echo "exec \$@";
-        fi
+        echo "su-exec $USER \$@";
     ) > ${BUILDDIR}/userwrapper.sh
 else
     (
@@ -156,11 +154,7 @@ else
         echo "usermod -a -G wheel $USER 2>/dev/null || :";
         echo "usermod -a -G adm $USER 2>/dev/null || :";
         echo "export HOME=/home/$USER" ;
-        if [[ 0${IS_ROOTLESS} == 0 ]]; then
-            echo "exec chroot --userspec=$USER / \$@";
-        else
-            echo "exec \$@";
-        fi
+        echo "exec chroot --userspec=$USER / \$@";
     ) > ${BUILDDIR}/userwrapper.sh
 fi
```

This way we will not create the user in the container: we don't need it and don't use it in fact.

BTW, please, keep the script Bourne shell compatible (without using bash specific features). `!#/bin/sh` is shebang, so it'll be executed with dash on Debian / Ubuntu.

Reply:
I will try to test this. Since I haven't used this software since Aug, I'm trying to remember how I tested etc.

Reply:
Done in b1d61a1